src/gui/annotator/AnnotationWidget.cpp

```diff
@@ -103,12 +103,12 @@ QImage AnnotationWidget::imageAt(int index) const
 void AnnotationWidget::loadImage(const QPixmap &pixmap)
 {
 	auto currentAnnotationArea = annotationArea();
-	removeTab(0);
-	addTab(pixmap, QString(), QString());
-	return;
+	auto currentIndex = mAnnotationTabWidget->currentIndex();
 
-	// unreachable code:
-	if(currentAnnotationArea == nullptr) {
+	if (currentIndex <= 0 && currentAnnotationArea == nullptr) {
```

Comment:
Why do you need those three cases now? Isn't it sufficient to check in this method only this:
```
if (currentIndey >= 0) 
    removeTab(currentIndex)
    insertTab(currentIndex)
else
    addTab()
```